packages/debug/src/browser/debug-session.tsx

```diff
@@ -379,8 +383,13 @@ export class DebugSession implements CompositeTreeElement {
 
     protected async runInTerminal({ arguments: { title, cwd, args, env } }: DebugProtocol.RunInTerminalRequest): Promise<DebugProtocol.RunInTerminalResponse['body']> {
         const terminal = await this.doCreateTerminal({ title, cwd, env });
-        terminal.sendText(args.join(' ') + '\n');
-        return { processId: await terminal.processId };
+        // TODO: Getting a property shouldn't map to a promise for an RPC call.
+        //       Either change the way the API works or transform the attribute
+        //       into a function.
+        // Parallelize promise resolution by requesting both now:
+        const { processId, processInfo } = terminal;
+        terminal.sendText(this.runInTerminalService.prepareCommandLine(await processInfo, { cwd, args, env }) + '\n');
```

Comment:
Could we encapsulate it in `terminal` instead of  `RunInTerminalService`?

i.e. a client will simply call `await terminal.sendCommand({ cwd, args, env });`?

Reply:
:thinking: while I really don't like writing commands in place of a user, we can do that since the code is there now anyway. Will do.